parcels/particle.py

```diff
@@ -148,6 +176,9 @@ def cptr(i):
             lon = np.linspace(start[0], finish[0], size, dtype=np.float32)
             lat = np.linspace(start[1], finish[1], size, dtype=np.float32)
 
+        if start_field is not None:
+            lon, lat = positions_from_density_field(size, getattr(self.grid, start_field))
```

Comment:
You are right about the initialisation checks, but most of them happen in the `Field` constructor, so ensuring the object is of type `parcels.Field` should be enough. And yes, passing the object rather than its name is definitely more pythonic. :snake: :wink: